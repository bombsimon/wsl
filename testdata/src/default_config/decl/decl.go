package testpkg

func fn1() {
	var a = 1
	var b = 2 // want "missing whitespace decreases readability"

	const c = 3
	const d = 4 // want "missing whitespace decreases readability"

	e := 5
	var f = 6 // want "missing whitespace decreases readability"
	g := 7 // want "missing whitespace decreases readability"

	_ = a
	_ = b
	_ = c
	_ = d
	_ = e
	_ = f
	_ = g
}

func fn2() {
	var a = 1
	var b = a // want "missing whitespace decreases readability"
}
