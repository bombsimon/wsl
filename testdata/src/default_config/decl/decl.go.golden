package testpkg

func fn1() {
 	// want +2 `missing whitespace above this line \(never cuddle decl\)`
	var (
		a = 1
		b = 2
	)

 	// want +2 `missing whitespace above this line \(never cuddle decl\)`
	const (
		c = 3
		d = 4
	)

	// want +3 `missing whitespace above this line \(never cuddle decl\)`
	// want +3 `missing whitespace above this line \(invalid statement above assign\)`
	e := 5

	var f = 6

	g := 7

	// want +2 `missing whitespace above this line \(never cuddle decl\)`
	var (
		a = 1
		b = a
	)
}

func fn2() {
	var x = func() { // want +1 `unnecessary whitespace \(leading-whitespace\)`
		return 1
	}()
}

func fn3() {
	// want +4 `missing whitespace above this line \(never cuddle decl\)`
	// want +4 `missing whitespace above this line \(never cuddle decl\)`
	// want +4 `missing whitespace above this line \(never cuddle decl\)`
	var (
		a = 1
		b = 2
		c = 3
		d = 4
	)
}

func fn4() {
	// want +5 `missing whitespace above this line \(never cuddle decl\)`
	var (
		a = 1
		b = 2
		c = 3
		d = 4
	)
}

func fn5() {
	// want +7 `missing whitespace above this line \(never cuddle decl\)`
	// want +7 `missing whitespace above this line \(never cuddle decl\)`
	// want +9 `missing whitespace above this line \(never cuddle decl\)`
	var (
		a = 1
		b = 2
		x int
		z = func() int {
			return 1
		}
		c = 3
		d = 4
	)
}

func fn6() {
	// want +5 `missing whitespace above this line \(never cuddle decl\)`
	// want +5 `missing whitespace above this line \(never cuddle decl\)`
	// want +5 `missing whitespace above this line \(never cuddle decl\)`
	// want +5 `missing whitespace above this line \(never cuddle decl\)`
	var (
		a = 1
		b = 2
	)

	var c = 3 // test

	var (
		d = 4
		e = 5
	)
}

func fn7() {
	// want +2 `missing whitespace above this line \(never cuddle decl\)`
	var (
		a = 1
		b = 2
	)

	var c = 3

	// want +2 `missing whitespace above this line \(never cuddle decl\)`
	var (
		d = 4
		e = 5
	)
}

func fn8() {
 	// want +3 `missing whitespace above this line \(never cuddle decl\)`
	// Comment above
	var (
		g = 7
		h = 8
	)
	// Comment after
}

func fn9() {
	type S1 struct {
		N int
	}

	type S2 struct { // want `missing whitespace above this line \(never cuddle decl\)`
		N int
	}
}

func fn10() {
 	// want +5 `missing whitespace above this line \(never cuddle decl\)`
 	// want +5 `missing whitespace above this line \(never cuddle decl\)`
 	// want +5 `missing whitespace above this line \(never cuddle decl\)`
	a := 1
	b := 2

	var (
		c int
		d        = "string"
		e string = "string"
	)

	f := 3 // want `missing whitespace above this line \(invalid statement above assign\)`
}

func fn11() {
 	// want +2 `missing whitespace above this line \(never cuddle decl\)`
	var (
		a int
		b int
	)

	if b > 0 { // want `missing whitespace above this line \(too many statements above if\)`
		_ = 1
	}
}

func fn12() {
 	// want +2 `missing whitespace above this line \(never cuddle decl\)`
	var a int

	var b int // Not grouped due to this comment
	if b > 0 {
		_ = 1
	}
}
